fenicsxconcrete/boundary_conditions/bcs.py

```diff
@@ -71,46 +80,54 @@ def add_dirichlet_bc(
             A hint which method should be used to locate the dofs.
             Choice: 'topological' or 'geometrical'.
         entity_dim : optional, int
-            The entity dimension in case `method=topological`.
+            The dimension of the entities to be located topologically.
+            Note that `entity_dim` is required if `sub` is not None and
+            `method=geometrical`.
         """
-        if boundary is None:
-            assert isinstance(value, dolfinx.fem.DirichletBCMetaClass)
+        if isinstance(value, dolfinx.fem.DirichletBCMetaClass):
             self._bcs.append(value)
         else:
             assert method in ("topological", "geometrical")
             V = self.V.sub(sub) if sub is not None else self.V
 
-            if method == "topological":
+            # dolfinx.fem.locate_dofs_geometrical(Vsub, plane_at) raises
+            # RuntimeError
```

Comment:
```suggestion
```
I have no objections to removing it. I just included it to explain line 95

Reply:
leave it

Reply:
```suggestion
            # if sub is not None and method=="geometrical"
            # dolfinx.fem.locate_dofs_geometrical(V, boundary) will raise a RuntimeError
            # because dofs of a subspace cannot be tabulated
```
maybe this is better understandable?

Reply:
hmm, somehow this is marked as outdated and therefore I also cannot commit the suggestion here. Will do it locally then.
If you are commenting on a line in the git diff here, the symbol for the suggestion is the one left to the "H".